cmd/apiserver/server.go

```diff
@@ -60,12 +57,87 @@ func (s *server) createCalculationBulk(c *gin.Context) {
 	}
 
 	if err := s.client.Create(s.ctx, bulk); err != nil {
-		responseError(c, "couldn't create calculation", err)
+		responseError(c, "couldn't create calculation bulk", err)
 	} else {
 		c.JSON(http.StatusOK, gin.H{"data": bulk})
 	}
 }
 
+func (s *server) createWorkerPool(c *gin.Context) {
+	body, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		responseError(c, "couldn't read body", err)
+		return
+	}
+
+	workerPool := &workersv1.WorkerPool{}
+
+	var NameBody struct {
+		Name string `json:"name,omitempty"`
+	}
+
+	if err := json.Unmarshal(body, &NameBody); err != nil {
+		responseError(c, "couldn't unmarshal body", err)
+	}
+
+	workerPool.Name = fmt.Sprintf("workerpool-%s", NameBody.Name)
+
+	if err := s.client.Get(s.ctx, ctrlruntimeclient.ObjectKey{Namespace: s.namespace, Name: workerPool.Name}, workerPool); err == nil {
```

Comment:
We don't need to get anything here.

Reply:
You already do that below. When you `.Create` it. You should be able to check if the error is the `IsAlreadyExists` error and respond to that, or just respond to any of the errors that the `Create` method will generate. If the worker pool already exists, the user will know.